Hey Folks, 

Next time you have a problems with scalability presubmits (or other CI jobs) please do notify our oncall - go.k8s.io/oncall (which happens to be me this week). In addition you can also ping us on #sig-scalability slack

I took a look at your PR and the hollow-nodes seem to be crashlooping with the following error:
```
E0226 12:21:17.143588       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 197 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x392b880, 0x6c3bf50)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x392b880, 0x6c3bf50)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).canAdmitPod(0xc00079ca80, 0x0, 0x0, 0x0, 0xc00020ec00, 0x0, 0x0, 0x6bd3e63, 0xa, 0x773)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1793 +0xdb
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).HandlePodAdditions(0xc00079ca80, 0xc00049b980, 0x2, 0x2)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:2070 +0x30b
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoopIteration(0xc00079ca80, 0xc0002e7da0, 0x4987c80, 0xc00079ca80, 0xc0009095c0, 0xc000909620, 0xc0007b3d40, 0xc00014fb01)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1912 +0x55a
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoop(0xc00079ca80, 0xc0002e7da0, 0x4987c80, 0xc00079ca80)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1855 +0x371
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).Run(0xc00079ca80, 0xc0002e7da0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1452 +0x27c
k8s.io/kubernetes/cmd/kubelet/app.startKubelet.func1()
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:1128 +0x3c
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0007e8460)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5e
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0007e8460, 0x48d9880, 0xc00068acf0, 0xc000518301, 0xc0000b04e0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xa3
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0007e8460, 0x0, 0x0, 0x1, 0xc0000b04e0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0xe2
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0007e8460, 0x0, 0xc0000b04e0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by k8s.io/kubernetes/cmd/kubelet/app.startKubelet
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:1127 +0x9a
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x323653b]

goroutine 197 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x392b880, 0x6c3bf50)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).canAdmitPod(0xc00079ca80, 0x0, 0x0, 0x0, 0xc00020ec00, 0x0, 0x0, 0x6bd3e63, 0xa, 0x773)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1793 +0xdb
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).HandlePodAdditions(0xc00079ca80, 0xc00049b980, 0x2, 0x2)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:2070 +0x30b
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoopIteration(0xc00079ca80, 0xc0002e7da0, 0x4987c80, 0xc00079ca80, 0xc0009095c0, 0xc000909620, 0xc0007b3d40, 0xc00014fb01)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1912 +0x55a
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoop(0xc00079ca80, 0xc0002e7da0, 0x4987c80, 0xc00079ca80)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1855 +0x371
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).Run(0xc00079ca80, 0xc0002e7da0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1452 +0x27c
k8s.io/kubernetes/cmd/kubelet/app.startKubelet.func1()
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:1128 +0x3c
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0007e8460)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5e
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0007e8460, 0x48d9880, 0xc00068acf0, 0xc000518301, 0xc0000b04e0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xa3
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0007e8460, 0x0, 0x0, 0x1, 0xc0000b04e0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0xe2
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0007e8460, 0x0, 0xc0000b04e0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by k8s.io/kubernetes/cmd/kubelet/app.startKubelet
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubelet/app/server.go:1127 +0x9a
```

My guess is that we have nil pointer deference in the canAdmitPod? Will you be able to debug it from there?

Unfortunately, we don't have these logs gathered by log exporter (I needed to manually examine a kubemark cluster built from your PR). I will open an issue to fix that (gathering logs on kubemark components panic).

--- a/pkg/features/kube_features.go
+++ b/pkg/features/kube_features.go
@@ -140,6 +140,7 @@ const (
 
 	// owner: @lmdaly
 	// alpha: v1.16
+	// beta: v1.18
 	//
 	// Enable resource managers to make NUMA aligned decisions
 	TopologyManager featuregate.Feature = "TopologyManager"
@@ -557,7 +558,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	AttachVolumeLimit:              {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.19
 	CPUManager:                     {Default: true, PreRelease: featuregate.Beta},
 	CPUCFSQuotaPeriod:              {Default: false, PreRelease: featuregate.Alpha},
-	TopologyManager:                {Default: false, PreRelease: featuregate.Alpha},
+	TopologyManager:                {Default: true, PreRelease: featuregate.Beta},
 	ServiceNodeExclusion:           {Default: false, PreRelease: featuregate.Alpha},
 	NodeDisruptionExclusion:        {Default: false, PreRelease: featuregate.Alpha},
 	CSIDriverRegistry:              {Default: true, PreRelease: featuregate.Beta},

--- a/pkg/kubelet/apis/config/validation/validation_test.go
+++ b/pkg/kubelet/apis/config/validation/validation_test.go
@@ -53,7 +53,6 @@ func TestValidateKubeletConfiguration(t *testing.T) {
 		HairpinMode:                 kubeletconfig.PromiscuousBridge,
 		NodeLeaseDurationSeconds:    1,
 		CPUCFSQuotaPeriod:           metav1.Duration{Duration: 100 * time.Millisecond},
-		TopologyManagerPolicy:       "none",
 	}
 	if allErrors := ValidateKubeletConfiguration(successCase1); allErrors != nil {
 		t.Errorf("expect no errors, got %v", allErrors)
@@ -86,7 +85,6 @@ func TestValidateKubeletConfiguration(t *testing.T) {
 		HairpinMode:                 kubeletconfig.PromiscuousBridge,
 		NodeLeaseDurationSeconds:    1,
 		CPUCFSQuotaPeriod:           metav1.Duration{Duration: 100 * time.Millisecond},
-		TopologyManagerPolicy:       "none",
 		ReservedSystemCPUs:          "0-3",
 	}
 	if allErrors := ValidateKubeletConfiguration(successCase2); allErrors != nil {
@@ -118,9 +116,8 @@ func TestValidateKubeletConfiguration(t *testing.T) {
 		HairpinMode:                 "foo",
 		NodeLeaseDurationSeconds:    -1,
 		CPUCFSQuotaPeriod:           metav1.Duration{Duration: 0},
-		TopologyManagerPolicy:       "",
 	}
-	const numErrsErrorCase1 = 26
+	const numErrsErrorCase1 = 25
 	if allErrors := ValidateKubeletConfiguration(errorCase1); len(allErrors.(utilerrors.Aggregate).Errors()) != numErrsErrorCase1 {
 		t.Errorf("expect %d errors, got %v", numErrsErrorCase1, len(allErrors.(utilerrors.Aggregate).Errors()))
 	}
@@ -152,7 +149,6 @@ func TestValidateKubeletConfiguration(t *testing.T) {
 		HairpinMode:                 kubeletconfig.PromiscuousBridge,
 		NodeLeaseDurationSeconds:    1,
 		CPUCFSQuotaPeriod:           metav1.Duration{Duration: 100 * time.Millisecond},
-		TopologyManagerPolicy:       "none",
 		ReservedSystemCPUs:          "0-3",
 	}
 	const numErrsErrorCase2 = 1

--- a/pkg/kubelet/cm/topologymanager/topology_manager.go
+++ b/pkg/kubelet/cm/topologymanager/topology_manager.go
@@ -18,6 +18,7 @@ package topologymanager
 
 import (
 	"fmt"
+	"sync"
 
 	"k8s.io/api/core/v1"
 	"k8s.io/klog"
@@ -54,6 +55,7 @@ type Manager interface {
 }
 
 type manager struct {
+	mutex sync.Mutex
 	//The list of components registered with the Manager
 	hintProviders []HintProvider
 	//Mapping of a Pods mapping of Containers and their TopologyHints
@@ -189,13 +191,18 @@ func (m *manager) AddHintProvider(h HintProvider) {
 }
 
 func (m *manager) AddContainer(pod *v1.Pod, containerID string) error {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
 	m.podMap[containerID] = string(pod.UID)
 	return nil
 }
 
 func (m *manager) RemoveContainer(containerID string) error {
-	klog.Infof("[topologymanager] RemoveContainer - Container ID: %v", containerID)
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
 
+	klog.Infof("[topologymanager] RemoveContainer - Container ID: %v", containerID)
 	podUIDString := m.podMap[containerID]
 	delete(m.podMap, containerID)
 	if _, exists := m.podTopologyHints[podUIDString]; exists {

